### ISSUE TITLE
Rounding mode should always round down in some cases

### ISSUE THREAD

**Issue by defuse** [zcash/zcash-android-wallet-sdk#32]
Currently the rounding mode is set to `HALF_EVEN` which can round a value up. This might cause usability problems, for example if the user wants to send their entire balance of ZEC, and the amount displayed in the UI is rounded-up, they might type the rounded-up value to send and get an error that they have insufficient funds. There is also rounding-up done as part of `BigDecimal.divide` in `convertUsdToZec` which may cause similar problems.

**Comment by gmale** [comment]
It seems like rounding down would have a similar potential for problems.

For example if the user wants to send their entire balance of ZEC, and the amount displayed in the UI is rounded-down, they might type the rounded-down value to send and get stuck with a trace amount of ZEC smaller than the miner's fee (and thereby impossible to remove from the wallet w/i injecting funds).

It doesn't seem that the rounding mode can fix UI issues. Instead, good UI design would have a "send all" button or detect and prompt a user who is in one of these edge cases and ask if they'd like to send all their funds, instead. Thereby, it might make sense for rounding to adhere to banker's rounding (half-even) so that the distribution of rounding errors is more even, overall. Another solution might be to make the default rounding mode in the SDK configurable.

Thoughts? @defuse

**Comment by defuse** [comment]
Good point, yeah simply rounding down doesn't solve the actual problem. Rounding down might also create problems when you're trying to pay an exact amount for some product/service and the seller's automated systems are pedantic and won't accept 0.00...01 ZEC less than they asked.

**Comment by gmale** [comment]
@defuse any thoughts on this?

**Comment by defuse** [comment]
Yeah this is probably out of scope of the SDK. It might be worth a note in documentation somewhere.

**Comment by geffenz** [comment]
I think we can sidestep this issue by having a MAX button on the send flow. That way the app knows they intend to send everything they can and don't have to type it.

**Comment by defuse** [comment]
A max button would be great! Gemini has a max button and I use it every time.
